### ISSUE TITLE
install oneof <single package> installs nothing

### ISSUE THREAD

**Issue by dmach** [openSUSE/libsolv#289]
I have found a bug in install oneof - when installing a single package, it installs nothing.
I've found it when using testsolv utility, I haven't verified it by using libsolv directly.

testcase.t:
```
# empty system repo (no installed packages)
repo @System 0 testtags <inline>
#>=Ver: 2.0

repo test 0 testtags <inline>
#>=Ver: 2.0
#>=Pkg: pkg 1 2 i686
#>=Pkg: pkg 1 2 x86_64

system x86_64 rpm @System
poolflags implicitobsoleteusescolors
solverflags allowdowngrade allowvendorchange keepexplicitobsoletes bestobeypolicy keeporphans yumobsoletes

# test1: works fine
job install pkg pkg-1-2.x86_64@test [setevr,setarch]
result transaction,problems <inline>
#>install pkg-1-2.x86_64@test

# test2: doesn't work
nextjob
job install oneof pkg-1-2.x86_64@test [setevr,setarch]
result transaction,problems <inline>
#>install pkg-1-2.x86_64@test

# test3: works (the same package listed twice)
nextjob
job install oneof pkg-1-2.x86_64@test pkg-1-2.x86_64@test [setevr,setarch]
result transaction,problems <inline>
#>install pkg-1-2.x86_64@test

# test4: works fine
nextjob
job install oneof pkg-1-2.x86_64@test pkg-1-2.i686@test [setevr,setarch]
result transaction,problems <inline>
#>install pkg-1-2.x86_64@test

# test5: works fine (reversed order of the RPMs)
nextjob
job install oneof pkg-1-2.i686@test pkg-1-2.x86_64@test [setevr,setarch]
result transaction,problems <inline>
#>install pkg-1-2.x86_64@test
```

**Comment by mlschroe** [comment]
Hmm yes. It's fortunately a bug in testsolv and not in the solver.

**Comment by mlschroe** [comment]
Fixed with commit 0302dce36cb5066a5c818d616a047dc93cfd025d. Thanks!
